keps/sig-auth/3257-trust-anchor-sets/README.md

```diff
@@ -0,0 +1,1328 @@
+the named trust anchor set of the named ClusterTrustBundle object, an use those
+for validating the certificate presented by the aggregated API server.
+kube-apiserver will follow updates to the ClusterTrustBundle object, and use the
+updated certificates within a few seconds.
+
+```diff
+type APIServiceSpec struct {
+    // ...
+
+    // InsecureSkipTLSVerify disables TLS certificate verification when communicating with this server.
+    // This is strongly discouraged.  You should use the CABundle instead.
+    InsecureSkipTLSVerify bool `json:"insecureSkipTLSVerify,omitempty" protobuf:"varint,4,opt,name=insecureSkipTLSVerify"`
+    // CABundle is a PEM encoded CA bundle which will be used to validate an API server's serving certificate.
+    // If unspecified, system trust roots on the apiserver are used.
+    // +listType=atomic
+    // +optional
+    CABundle []byte `json:"caBundle,omitempty" protobuf:"bytes,5,opt,name=caBundle"`
++	// A reference to a dynamic trust anchor set that will be used to validate
++	// the webhook's server certificate.
++	//
++	// Mutually exclusive with CABundle and InsecureSkipTLSVerify.
++	//
++	// This field is alpha.  Using it requires setting the ClusterTrustBundle
++	// feature gate.
++	//
++	// +optional
++	TrustAnchors *TrustAnchorSetReference `json:"trustAnchors,omitempty"`
+
+    // ...
+}
++
++// TrustAnchorSetReference names a particular trust anchor set within a
++// ClusterTrustBundle object.
++type TrustAnchorSetReference struct {
++	// The ClusterTrustBundle object to refer to.
++	ClusterTrustBundleName string `json:"clusterTrustBundleName"`
++
++	// The individual named trust anchor set within the ClusterTrustBundle.
++	TrustAnchorSetName string `json:"trustAnchorSetName"`
++}
+```
+
+### trustAnchors Projected Volume Source
+
+The trustAnchors projected volume source is responsible for retrieving a
+designated trust anchor set from a signer and publishing it into the container
+filesystem.
+
+All of the new fields described in this section are gated by the
+ClusterTrustBundle kubelet feature gate.  Unless the feature gate is enabled,
+usage of these alpha objects and fields will be rejected by kubelet.
+
+#### Configuration Object Definition
+
+```go
+// TrustAnchorsProjection is the top-level configuration object for
+// instances of the trustAnchors kubelet projected volume source.
+type TrustAnchorsProjection struct {
+  // Which trust anchor sets should we retrieve from the signer, and where
+  // should we write them into the volume?
+  Files []TrustAnchorSetFile `json:"files"`
+}
+
+// TrustAnchorsFile is a oneof that holds different strategies for writing
+// trust anchor sets to the projected volume.
+//
+// Conceptually, it’s a oneof, but only the FlatFile option is provided
+// for now.  In the future, different formats that can hold more than one
+// named trust anchor set may be supported.  Or we might support specialty
+// formats like Java keystores.
```

Comment:
I believe you need a discriminator field.  See #3377 and the related [discussion thread](https://groups.google.com/g/kubernetes-api-reviewers/c/Q7UC6hptRlw/m/FRwY84rnAQAJ).

Reply:
This whole field has been removed, so this is moot.